cmd/node_agent/na/aws.go

```diff
@@ -41,16 +68,44 @@ func (na *awsPlatformImpl) IsProperPlatform() bool {
 	return na.client.Available()
 }
 
-// Extract service identity from userdata. This function should be
-// pluggable for different AWS deployments in the future.
+// This function should be pluggable for different AWS deployments.
 func (na *awsPlatformImpl) GetServiceIdentity() (string, error) {
 	return "", nil
 }
 
+func (na *awsPlatformImpl) GetInstanceIdentityDocument() ([]byte, error) {
```

Comment:
Do you want to make this function private?

Reply:
will do

Reply:
I'll change it in next PR